ambry-replication/src/main/java/com/github/ambry/replication/ReplicationManager.java

```diff
@@ -381,7 +384,10 @@ public void onPartitionBecomeOfflineFromInactive(String partitionName) {
       }
       // set local store state to OFFLINE and initiate disconnection
       store.setCurrentState(ReplicaState.OFFLINE);
-      replicaSyncUpManager.initiateDisconnection(localReplica);
+      if (!store.isEmpty()) {
+        // Wait for peers to sync up if this replica is not empty
+        replicaSyncUpManager.initiateDisconnection(localReplica);
```

Comment:
nit: not related to this pr, but the initiateDisconnection and initiateDeactivation looks a little bit confused to me that I thought every store need to go through this process when it has been marked as OFFLINE. However, it seems like only the non-empty store would need to go through this process in order to make sure peer replica catch up everything. Can we add the comment in the interface side to make it more clear?

Reply:
Sure. Added comments that we don't need this step for empty replica.